app/processing/statement/get-statement.js

```diff
@@ -0,0 +1,26 @@
+const { getDetails, getAdress, getDetailedFunding, getScheme, getDetailedPayments } = require('./components')
+const getCalculation = require('../calculation')
+const getPaymentRequest = require('../payment-request')
+const getSettlement = require('../settlement')
+
+const getStatement = async (settlementId) => {
+  const settlement = await getSettlement(settlementId)
+  const paymentRequestId = settlement.paymentRequestId
+  const paymentRequest = await getPaymentRequest(paymentRequestId)
+  const calculation = await getCalculation(paymentRequestId)

```

Comment:
I don't think this will work as the `paymentRequestId` is null when data is sent to it.  This is because the data can arrive in any order and even if calculation comes in after payment request there is nothing that ever sets the `paymentRequestId`.

What we need to do is pass a reverse engineered invoice number to this function and instead match by invoice number.

There is a function for reverse engineering the invoice number prepared for this, but it is incomplete (but not far off) `app/processing/reverse-engineer-invoice-number.js`.

To reverse engineer the invoice number you'd need to do this.

1. Payment request invoice -> `S000000100000001V001`
2. Take first 8 characters to get this `S0000001`
3. Replace `S` with `SFI` to get this `SFI0000001`

That's it.  When it comes to SFI Pilot and Lump Sums the S needs to be replaced with something else, but a problem for another day.